rocAL/rocAL/source/pipeline/node.cpp

```diff
@@ -70,3 +78,25 @@ Node::update_parameters()
 {
     update_node();
 }
+
+// Create vx_tensor for the crop coordinates
+void Node::create_crop_tensor(vx_tensor &crop_tensor, void **crop_coordinates) {
+    vx_size num_of_dims = 2;
+    vx_size stride[num_of_dims];
+    std::vector<size_t> crop_tensor_dims = {_batch_size, 4};
+    if(_inputs[0]->info().layout() == RocalTensorlayout::NFCHW || _inputs[0]->info().layout() == RocalTensorlayout::NFHWC)
+        crop_tensor_dims = {_inputs[0]->info().dims()[0] * _inputs[0]->info().dims()[1], 4}; // For Sequences pre allocating the ROI to N * F to replicate in OpenVX extensions
+    stride[0] = sizeof(vx_uint32);
+    stride[1] = stride[0] * crop_tensor_dims[0];
+    vx_enum mem_type = VX_MEMORY_TYPE_HOST;
+    if (_inputs[0]->info().mem_type() == RocalMemType::HIP)
+        mem_type = VX_MEMORY_TYPE_HIP;
+    allocate_host_or_pinned_mem(crop_coordinates, stride[1] * 4, _inputs[0]->info().mem_type());
+
+    crop_tensor = vxCreateTensorFromHandle(vxGetContext((vx_reference) _graph->get()), num_of_dims, crop_tensor_dims.data(), VX_TYPE_UINT32, 0, 
+                                                                  stride, (void *)(*crop_coordinates), mem_type);
+    vx_status status;
+    if ((status = vxGetStatus((vx_reference)crop_tensor)) != VX_SUCCESS)
+        THROW("Error: vxCreateTensorFromHandle(crop_tensor: failed " + TOSTR(status))
+    return;
```

Comment:
do we need a return statement here since create_crop_tensor has return_type void